File: src/main/java/com/machfour/macros/objects/FoodCategory.java
package com.machfour.macros.objects;

import com.machfour.macros.core.*;

public class FoodCategory extends MacrosEntityImpl<FoodCategory> {

    public FoodCategory(ColumnData<FoodCategory> data, ObjectSource objectSource) {
        super(data, objectSource);
    }

    @Override
    public String toString() {
        return getName();
    }

    @Override
    public Table<FoodCategory> getTable() {
        return table();
    }

    public String getName() {
        return getData(Schema.FoodCategoryTable.NAME);
    }

    public static Factory<FoodCategory> factory() {
        return FoodCategory::new;
    }

    public static Table<FoodCategory> table() {
        return Schema.FoodCategoryTable.instance();
    }
    @Override
    public Factory<FoodCategory> getFactory() {
        return factory();
    }
}

File: src/main/kotlin/com/machfour/macros/objects/FoodCategory.kt
package com.machfour.macros.objects

import com.machfour.macros.core.*

class FoodCategory(data: ColumnData<FoodCategory>, objectSource: ObjectSource) : MacrosEntityImpl<FoodCategory>(data, objectSource) {
    companion object {
        val factory: Factory<FoodCategory> = Factory { dataMap, objectSource -> FoodCategory(dataMap, objectSource) }
        val table: Table<FoodCategory>
            get() = Schema.FoodCategoryTable.instance
    }

    override val factory: Factory<FoodCategory>
        get() = Companion.factory

    override val table: Table<FoodCategory>
        get() = Companion.table

    val name: String
        get() = getData(Schema.FoodCategoryTable.NAME)!!

    override fun toString(): String {
        return name
    }
}
